src/aten/sycl/UnaryKernels.cpp

```diff
@@ -18,6 +18,72 @@ struct AbsFunctor {
   }
 };
 
+template <typename scalar_t>
+struct SinFunctor {
+  scalar_t operator()(const scalar_t a) const {
+    return std::sin(a);
+  }
+};
+
+template <typename scalar_t>
+struct CosFunctor {
+  scalar_t operator()(const scalar_t a) const {
+    return std::cos(a);
+  }
+};
+
+template <typename scalar_t>
+struct LogFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return std::log(a);
+  }
+};
+
+template <typename scalar_t>
+struct SqrtFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return std::sqrt(a);
+  }
+};
+
+template <typename scalar_t>
+struct RsqrtFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return sycl::rsqrt(float(a));
```

Comment:
May I know why need to cast `a` to `float` when `a` is not `double` or `complex` data type?

Reply:
Align with cuda for acc consideration.